openquake/hazardlib/gsim/base.py

```diff
@@ -1009,3 +1012,502 @@ def __getitem__(self, imt):
+    """
+    Convert an hdf5 group containins only data sets to a dictionary of
+    data sets
+    :param hdfgroup:
+        Instance of :class: h5py.Group
+    :returns:
+        Dictionary containing each of the datasets within the group arranged
+        by name
+    """
+    return dict([(key, hdfgroup[key][:]) for key in hdfgroup.keys()])
+
+
+class AmplificationTable(object):
+    """
+    Class to apply amplification from the GMPE tables.
+
+    :param shape:
+        Shape of the amplification arrays as a tuple of (Number Distances,
+        Number IMTs, Number Magnitudes, Number Amplification Levels)
+    :param periods:
+        Spectral periods defined in table
+    :param mean:
+        Amplification factors for the mean ground motion
+    :param sigma:
+        List of modification factors for the standard deviation of ground
+        motion
+    :param magnitudes:
+        Magnitude values for the tables
+    :param distances:
+        Distance values for the tables
+    :param parameter:
+        Parameter to which the amplification applies. Must be an element
+        inside the __slots__ defines in the :class: openquake.hazardlib.
+        gsim.base.RuptureContext or the :class: openquake.hazardlib.gsim.base.
+        SitesContext
+    :param values:
+        Array of values to which each amplification table corresponds
+    :param element:
+        Indicates if the amplification corresponds to a rupture attribute or
+        a site attribute
+    """
+    def __init__(self, amplification_group, magnitudes, distances):
+        """
+        Setup the amplification factors.
+
+        :param amplification_group:
+            Amplification model as instance of :class: h5py.Group
+        :param magnitudes:
+            Array of magnitudes
+        :param distances:
+            Array of distances
+        """
+        self.shape = None
+        self.periods = None
+        self.mean = None
+        self.sigma = None
+        self.magnitudes = magnitudes
+        self.distances = distances
+        self.parameter = amplification_group.attrs["apply_to"]
+        self.values = numpy.array([float(key)
+                                   for key in amplification_group.keys()])
+        self.argidx = numpy.argsort(self.values)
+        self.values = self.values[self.argidx]
+        if self.parameter in RuptureContext.__slots__:
+            self.element = "Rupture"
+        elif self.parameter in SitesContext.__slots__:
+            self.element = "Sites"
+        else:
+            raise ValueError("Amplification parameter %s not recognised!"
+                             % self.parameter)
+        self._build_data(amplification_group)
+
+    def _build_data(self, amplification_group):
+        """
+        Creates the numpy array tables from the hdf5 tables
+        """
+        # Determine shape of the tables
+        n_levels = len(amplification_group)
+        n_d, n_p, n_m = amplification_group.items()[0][1]["IMLs/SA"].shape
```

Comment:
Perhaps you can use an intermediate name to make it clear what `amplification_group.items()[0][1]` is, or use named constants for the numbers 0 and 1 so that it is clear what you are extracting from the items.